src/beamlime/services/dashboard.py

```diff
@@ -379,14 +380,31 @@ def maybe_unit(dim: str) -> str:
             'showlegend': False,
         }
         y_size, x_size = data.shape
-        if y_size < x_size:
-            fig.update_layout(width=size, **opts)
-            fig.update_yaxes(scaleanchor="x", scaleratio=1, constrain="domain")
-            fig.update_xaxes(constrain="domain")
+        if data.coords[x_dim].unit is not None and (
+            maybe_unit(y_dim) == maybe_unit(x_dim)
+        ):
+            if y_size < x_size:
+                fig.update_layout(width=size, **opts)
+                fig.update_yaxes(scaleanchor="x", scaleratio=1, constrain="domain")
+                fig.update_xaxes(constrain="domain")
+            else:
+                fig.update_layout(height=size, **opts)
+                fig.update_xaxes(scaleanchor="y", scaleratio=1, constrain="domain")
+                fig.update_yaxes(constrain="domain")
         else:
-            fig.update_layout(height=size, **opts)
-            fig.update_xaxes(scaleanchor="y", scaleratio=1, constrain="domain")
-            fig.update_yaxes(constrain="domain")
+            # Set size based on pixel count
+            long = max(y_size, x_size)
+            short = min(y_size, x_size)
+            ratio = long / short
+            max_size = 900
+            if ratio > 3:
+                if y_size < x_size:
+                    fig.update_layout(width=max_size, height=max_size // 3, **opts)
+                else:
+                    fig.update_layout(width=max_size // 3, height=max_size, **opts)
+            else:
+                scale = max_size / long
+                fig.update_layout(width=x_size * scale, height=y_size * scale, **opts)
```

Comment:
All the plot sizing is a hack, and I am sure this made it less readable in other cases. Someone needs a better idea, sometime.